src/main/scala/com/fulcrumgenomics/bam/ReadAndRefPosIterator.scala

```diff
@@ -0,0 +1,202 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2022 Fulcrum Genomics
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.bam
+
+import com.fulcrumgenomics.FgBioDef.yieldAndThen
+import com.fulcrumgenomics.alignment.CigarElem
+import com.fulcrumgenomics.bam.ReadAndRefPosIterator.ReadAndRefPos
+import com.fulcrumgenomics.bam.api.SamRecord
+import htsjdk.samtools.util.CoordMath
+
+/** An iterator for each _mapped_ read base (i.e. has a corresponding reference base), with each value being the
+  * 1-based position in the read and reference respectively.
+  *
+  * Insertions and deletions will not be returned.  For example, an insertion consumes a read base, but has no
+  * corresponding reference base.  A deletion consumes a reference base, but has no corresponding read base.
+  *
+  * @param rec the record over which read and reference positions should be returned
+  * @param startReadPos the 1-based start position in the read
+  * @param endReadPos the 1-based inclusive end position in the read
+  * @param startRefPos the 1-based start position in the reference
+  * @param endRefPos the 1-based inclusive end position in the reference
```

Comment:
I think these need to be clearly documents as filters on what is returned by the iterator, i.e. it's not up to the user to calculate these correctly, just to use them to limit the set of values returned by the iterator, right?
Should they then be renamed to `[min|max][Read|Ref]pos`?  And perhaps have default values of 1 and MaxInt?